src/python/pants/pantsd/pants_daemon.py

```diff
@@ -204,17 +267,47 @@ def _run(self):
     self._setup_services(self._services)
     self._run_services(self._services)
 
-  def pre_fork(self):
-    """Pre-fork() callback for ProcessManager.daemonize()."""
-    for service in self._services:
-      service.pre_fork()
-
-    # Teardown the RunTracker's SubprocPool pre-fork.
-    RunTracker.global_instance().shutdown_worker_pool()
-    # TODO(kwlzn): This currently aborts tracking of the remainder of the pants run that launched
-    # pantsd.
-
   def post_fork_child(self):
-    """Post-fork() child callback for ProcessManager.daemonize()."""
-    self._native.set_panic_handler()
-    self._run()
+    """Post-fork() child callback for ProcessManager.daemon_spawn()."""
+    entry_point = '{}:launch'.format(__name__)
+    exec_env = combined_dict(os.environ, dict(PANTS_ENTRYPOINT=entry_point))
+    # Pass all of sys.argv so that we can proxy arg flags e.g. `-ldebug`.
+    cmd = [sys.executable] + sys.argv
+    self._logger.debug(
+      'cmd is: PANTS_ENTRYPOINT={} {}'
+      .format('{}:launch'.format(__name__), ' '.join(cmd))
+    )
+    p = subprocess.Popen(cmd, env=exec_env)
+    p.wait()
```

Comment:
Should we error if this fails? Why are we `Popen`+`wait`ing rather than `check_call`ing?

Reply:
thought about this prior and it would be nice to have, but:

1) failure to launch the daemon is expected to be rare
2) this is currently called via `ProcessManager.daemon_spawn()` which does a double fork to daemonize before calling this method, so we wouldn't be able to get the result of the `check_call` back
3) the daemon doesn't currently provide sensible exit codes

will add a TODO tho.